moulinette/core.py

```diff
@@ -442,16 +442,21 @@ def acquire(self):
         start_time = time.time()
 
         while True:
-            lock_pid = self._lock_PID()
 
-            if lock_pid is None:
+            lock_pids = self._lock_PIDs()
+
+            if self._is_son_of(lock_pids):
+                return
```

Comment:
Why use `return` here and `break` below ?

Reply:
Because if we are a son of the locked process, we don't want to actually acquire the lock.

Whereas (in the next lines), if there are no current locks, we want to take the lock (with `self._lock()`)